components/SignInV2.js

```diff
@@ -0,0 +1,220 @@
+            {label || <FormattedMessage defaultMessage="Continue with your email" />}
+          </Flex>
+          <Flex fontWeight={400} fontSize="16px" color="black.700" mb="50px" justifyContent="center">
+            {label || <FormattedMessage defaultMessage="Sign in or create a personal account to continue" />}
+          </Flex>
+          {!this.state.unknownEmail ? (
+            <React.Fragment>
+              <Container fontWeight={600} fontSize="13px" alignItems="left" mb="4px" width="100%">
+                {label || <FormattedMessage id="Form.yourEmail" defaultMessage="Your email address" />}
+              </Container>
+              <Container
+                as="form"
+                method="POST"
+                noValidate
+                data-cy="signIn-form"
+                onSubmit={event => {
+                  event.preventDefault();
+                  onSubmit(email);
+                  this.setState({ unknownEmail: this.props.unknownEmail });
+                }}
+              >
+                <StyledInput
+                  error={!!error}
+                  fontSize="14px"
+                  id="email"
+                  name="email"
+                  minWidth={120}
+                  onChange={({ target }) => {
+                    onEmailChange(target.value);
+                    // Feel free to remove the setTimeout when that issue is fixed
+                    // https://bugzilla.mozilla.org/show_bug.cgi?id=1524212
+                    setTimeout(() => {
+                      this.setState({ error: target.validationMessage, showError: false });
+                    }, 0);
+                  }}
+                  onKeyDown={e => {
+                    // See https://github.com/facebook/react/issues/6368
+                    if (e.key === ' ') {
+                      e.preventDefault();
+                    }
+                  }}
+                  onBlur={() => this.setState({ showError: true })}
+                  onInvalid={event => {
+                    event.preventDefault();
+                    this.setState({ error: event.target.validationMessage });
+                  }}
+                  placeholder="e.g., yourname@yourhost.com"
+                  required
+                  value={email}
+                  type="email"
+                  width={1}
+                />
+                {error && showError && (
+                  <Span display="block" color="red.500" pt={2} fontSize="10px" lineHeight="14px" aria-live="assertive">
+                    {error}
+                  </Span>
+                )}
+                <Flex justifyContent="center" mb="24px" mt="26px">
+                  <StyledButton
+                    data-cy="signin-btn"
+                    buttonStyle="primary"
+                    height={32}
+                    lineHeight={0}
+                    fontWeight="500"
+                    disabled={!email || error}
```

Comment:
Here the disabling is done only when the email field is empty (and the user touched it) or there's an error in the email that is entered (which is shown below the email field). If we don't disable we will see the button being enabled for these two obvious cases (see screenshots below). To me it doesn't make sense to enable the button when the email field is empty; nor when the email field has something like one character which is obviously not an email address and we show the validation error below the field. For all other errors we show the toast after the user clicks on the `Continue` button. But if you disagree, I am happy to change this behavior to let the user click on the `Continue` regardless even in these two cases. 

![image](https://user-images.githubusercontent.com/12435965/171550851-8d0e76ee-24a4-4b57-b101-6c6d8fabba34.png)

![image](https://user-images.githubusercontent.com/12435965/171550885-b72f0f0b-b7f1-4c96-b934-2082be322b8c.png)

Reply:
Ref: https://uxplanet.org/disabled-buttons-in-user-interface-4dafda3e6fe7

> It is still possible to disable a submit button by default for login, shipping, billing, and many other forms when all fields are empty. However, when the user provides the information, it’s better to enable the Submit button. Let users make mistakes and provide appropriate feedback so that they can easily recover from error states.

So disabled if empty, yes. But I would enable it even if the email is not valid.

```suggestion
                    disabled={!email}
```